mlir/lib/Quantum/Transforms/StaticCustomPatterns.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define DEBUG_TYPE "static-custom"
+
+#include "mlir/IR/PatternMatch.h"
+
+#include "Quantum/IR/QuantumOps.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "llvm/Support/Debug.h"
+
+using llvm::dbgs;
+using namespace mlir;
+using namespace catalyst::quantum;
+
+namespace {
+
+struct LowerStaticCustomOp : public OpRewritePattern<StaticCustomOp> {
+    using OpRewritePattern::OpRewritePattern;
+
+    LogicalResult matchAndRewrite(StaticCustomOp op, PatternRewriter &rewriter) const override
+    {
+        LLVM_DEBUG(dbgs() << "Lowering the following static custom operation:\n" << op << "\n");
+        SmallVector<Value, 4> paramValues;
+        auto staticParams = op.getStaticParams();
+        for (auto param : staticParams) {
```

Comment:
Oh Actually `op.getStaticParams() ` returns a `llvm::ArrayRef<double>` instead of the attribute which is convenient for looping over :) to get the attribute one should use op.getStaticParamsAttr();

Reply:
Oh neat, thanks :)